libsolidity/formal/CHC.cpp

```diff
@@ -749,7 +750,12 @@ void CHC::externalFunctionCall(FunctionCall const& _funCall)
 		return;
 	}
 
-	FunctionType const& funType = dynamic_cast<FunctionType const&>(*_funCall.expression().annotation().type);
+	FunctionCallOptions const* callOptions = nullptr;
+	Expression const* callExpr = &_funCall.expression();
+	if ((callOptions = dynamic_cast<FunctionCallOptions const*>(callExpr)))
```

Comment:
I'd either move the `callOptions` variable into the `if` or move the assignment out of the `if`. Current version is the worst of both worlds in my opinion :) Initially I mistook it for `==`.

```suggestion
	Expression const* callExpr = &_funCall.expression();
	auto const* callOptions =  dynamic_cast<FunctionCallOptions const*>(callExpr);
	if (callOptions)
```

Same in `SMTEncoder.cpp`.

Reply:
The idea was to update and test it at the same time, saving some lines :p because it might also be `nullptr`, in which case we don't want to update `callExpr`.

Reply:
But the part inside `if` will always be executed, won't it? It was actually less efficient because of the extra `nullptr` assignment :)

Reply:
Why would it always execute?

Reply:
Because it can't be tested without peforming the assignment.

Maybe I'm misunderstading what you meant but I'm just saying that this:
```c++
auto const* x = nullptr;
if (x = f())
```
is equivalent to this:
```c++
auto const* x = nullptr;
x = f();
if (x)
```
the assignment and `f()` get executed in both cases.

Reply:
Ah right, I misread "inside the if's parenthesis" and thought you meant the branch.

Reply:
I'll move the cast out of the if's condition.